biothings_explorer/call_apis_v2/query.py

```diff
@@ -83,13 +92,19 @@ def get_edge(self, metakg_edge, input_id):
 
         return format_response(resp_data, metakg_edge)
 
-    def get_edges(self, metakg_edge, input_ids):
+    def get_edges(self, metakg_edge, input_ids, batch_size=1000):
         query_operation = metakg_edge["bte"]["query_operation"]
-        if query_operation["support_batch"]:
-            combined_ids = ",".join(input_ids)
-            return self.get_edge(metakg_edge, combined_ids)
-
-        results = []
-        for input_id in input_ids:
-            results += self.get_edge(metakg_edge, input_id)
-        return results
+
+        for i in range(0, len(input_ids), batch_size):
+            start_index = batch_size * i
+            end_index = batch_size * (i + 1)
+            sub_input_ids = input_ids[start_index:end_index]
+
+            if query_operation["support_batch"]:
+                combined_ids = ",".join(sub_input_ids)
+                yield self.get_edge(metakg_edge, combined_ids)
```

Comment:
if we return it as a generator, we should make it return one edge a time, not a batch a time. That means `get_edge` should also be a generator, because one input_id can return multiple edges.

Reply:
got it, let me update.

Reply:
It's updated
@newgene

Reply:
well, `results` in `get_edges` is still a batch get yielded, right. It should yield one edge at a time as well.

Reply:
hum, ok, let me change to one edge at a time

Reply:
@newgene it's updated.

Reply:
@sengineer0 still not right here, `yield get_edge` only returns the generator itself, not the each edge yielded from `get_edge`.

Reply:
It's updated.
@newgene

Reply:
ok, `get_edge` works as expected now, `get_edges` should be updated to return a generator of each edge as well. @sengineer0

Reply:
oh, you want even get_edges must return one edge at a time.
let me see how to do that.